webapp/src/js/bootstrapper/purger.js

```diff
@@ -7,6 +7,11 @@ const META_BATCHES = 10; // purge 10 * 100 documents on every startup
 
 const sortedUniqueRoles = roles => JSON.stringify([...new Set(roles)].sort());
 const purgeFetch = (url) => {
+  if (!opts) {
```

Comment:
Missing `opts` doesn't mean purge is not enabled. The `opts` property here should contain `POUCHDB_OPTIONS` (like the host URL and such) and has absolutely nothing to do with actual purge settings. 

Missing `opts` means a major flaw in the app, because we set opts here: 
https://github.com/medic/cht-core/blob/master/webapp/src/js/bootstrapper/index.js#L228

Please remove this code, I think it's very risky to add code here for a testing page.

Reply:
Hi @dianabarsan, Yep, I saw where the `opts` is set, but if I login as Admin then it will exit early in this [line](https://github.com/medic/cht-core/blob/2c74a6e25428f18c7f41bd80880cbe789953f6a9/webapp/src/js/bootstrapper/index.js#L205) and the `opts` will be `undefined`, so when running the purge through the Testing page, it is throwing an exception and not running the purge anyways, also the app doesn't give any hint that the purge was successful. 
I don't think this code is dangerous and it's handling an exception that was breaking the purge. 
Maybe we can change the message text and remove the alert if it's too much, but leaving the rejection with an appropriate message so it'll be handled respectively (currently the catch is printing the message in the console).

Reply:
Unless we want to proceed purging even thought the `opts` is `undefined` and regardless of the user role (admin), then we sill need to fix a bit the code to prevent the exception.
This was part of my question regarding the requirement :)

Reply:
Maybe the testing page should check if the user is an admin and not enable purging?

Reply:
Purging for not enabled for online users, btw.

Reply:
Done! Fixed